sys/usb/usbus/cdc/acm/cdc_acm_stdio.c

```diff
@@ -76,4 +115,7 @@ void usb_cdc_acm_stdio_init(usbus_t *usbus)
 {
     usbus_cdc_acm_init(usbus, &cdcacm, _cdc_acm_rx_pipe, NULL,
                        _cdc_tx_buf_mem, sizeof(_cdc_tx_buf_mem));
+#ifdef MODULE_ARDUINO_BOOTLOADER
+    usbus_cdc_acm_set_coding_cb(&cdcacm, _coding_cb);
+#endif
```

Comment:
How about we modify this to:
```C
#ifdef USBUS_CDC_ACM_STDIO_DEFAULT_CODING_CB
    extern USBUS_CDC_ACM_STDIO_DEFAULT_CODING_CB
    usbus_cdc_acm-set_coding_cb(&cdcacm, USBUS_CDC_ACM_STDIO_DEFAULT_CODING_CB);
#endif
```
Optionally we can move the `extern USBUS_CDC_ACM_STDIO_DEFAULT_CODING_CB` to the top of the file.

This way it remains board independent and we can extend it later with other boards that require special coding callback handling besides the Arduino bossa bootloaders.

The Arduino code then only has to define this define somewhere in the Makefiles (or maybe in the board header and include that here)

Reply:
Not exactly what you are asking but I added the management of the board reset in a separate module.
I'm not sure the extern strategy you proposed is working since extern requires the complete function definition. Let me know what you think of the new approach.